File: hanly_koffman/chapter_05_repetition_and_loops/5.5_programming_exercise_1/main.c
/*
 * Author: C. Maxwell Reilly
 * Date: Monday, May 25, 2023
 */

#include<stdio.h>
#include<math.h>

#define PI 3.141562

double  sine, cosine;
int     degree, init_degree, final_degree, step_degree;


int
main(void)
{
    /* Variable declaration */

    init_degree = 0;
    final_degree = 360;
    step_degree = 10;

    /* Display the table heading */
    printf("    Degrees     sine     cosine\n");

    /* Display the table */
    for     (degree = init_degree;
            degree <= final_degree;
            degree += step_degree){
        sine = sin(degree * (PI / 180));
        cosine = cos(degree * (PI / 180));
        printf("%6c%3d%8c%7.2f%8c%7.2f\n", ' ', degree, ' ', sine, ' ', cosine);
    }
    return(0);

}

File: hanly_koffman/chapter_05_repetition_and_loops/5.5_programming_exercise_1/main.py
from math import sin, cos, pi

def main():
    # Variable declaration
    init_degree = 0
    final_degree = 360
    step_degree = 10

    # Display heading
    print("    Degrees    sine    cosine    ")
    i = init_degree
    while i <= final_degree:
        print("{blanks}{i}{blanks}{sine:.2f}{blanks}{cosine:.2f}".format(blanks = "    ", i = i, sine = sin(i * (pi/180)), cosine = cos(i * (pi/180))))
        i += step_degree
if __name__ == "__main__":
    main()
